### PR TITLE
Add some basic fuzzing

### PR DESCRIPTION
**Pull request by jeffa5**
One target tries loading from arbitrary bytes to check that we don't panic.

The other target tries performing some limited actions, then saving and loading the document to check that it doesn't mess with things.

### CODE REVIEW

File: automerge/Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,7 @@ license = "MIT"
 [features]
 optree-visualisation = ["dot"]
 wasm = ["js-sys", "wasm-bindgen"]
+arbitrary = ["dep:arbitrary", "smol_str/arbitrary"]
```
Comment (jeffa5) [inline automerge/Cargo.toml:12]: Note that this currently requires a nightly version of `cargo`, though it is being stabilised in the next rust release (1.60.0) https://github.com/rust-lang/rust/pull/94817 as namespaced-features.

**Comment by alexjg** [comment]
Interesting, do we panic on the load fuzzer? Quite a few of the existing decoders etc. are not fallible when they should be so we probably do panic, but it might be quite hard to actually expose those panics with a fuzzer I guess.

**Comment by jeffa5** [comment]
The load target gave me a panic which led to https://github.com/automerge/automerge-rs/pull/317

File: automerge/Cargo.toml (hunk 1 above)
Comment (alexjg) [reply automerge/Cargo.toml:12]: This is being released in two weeksright? Do we wait until then to merge the PR?

Comment (jeffa5) [reply automerge/Cargo.toml:12]: I was thinking we probably could.

**Comment by orionz** [comment]
Hey @jeffa5 - can we merge this, close it, or make it a draft?  What's the current state of affairs?

**Comment by jeffa5** [comment]
@orionz Now rebased, happy to merge if it looks good to you

**Comment by orionz** [comment]
this hasnt been touched in ages - happy to reopen it if you want to get it merged

**Comment by orionz** [comment]
this hasnt been touched in ages - happy to reopen it if you want to get it merged

**Comment by orionz** [comment]
this hasnt been touched in ages - happy to reopen it if you want to get it merged

**Pull request closed without merging**